src/db_field.ml

```diff
@@ -12,6 +12,17 @@ type t =
   | Date of Time.t
 [@@deriving sexp]
 
+let try_decode_exn ~src str =
+  let decoder = Encoding.decoder src in
+  let i = ref 0 in
+  while !i < String.length str do
+    match Encoding.decode decoder str !i (String.length str - !i) with
```

Comment:
Why not `let` bind `String.length`. It looks like you're computing the length twice per loop iteration?